### PR TITLE
Fix energy history updates

### PR DESCRIPTION
**Pull request by codyc1515**
Pulls the history from the Contact Energy API and displays this daily.

### CODE REVIEW

File: custom_components/contact_energy/sensor.py [hunk 1]
```diff
@@ -196,85 +244,15 @@ def make_attribute(self, date, today, response):
...
-                    else:
-                        self._state_attributes[condition] = data.get(condition, None)
-            spot_price_data = self._api.get_spot_price()
-            if spot_price_data:
-                _LOGGER.debug('Fetching daily prices...')
-                today = datetime.now().replace(hour=0, minute=0, second=0, microsecond=0)
-                todaysData = []
-                tomorrowsData = []
+    
                 yesterdaysData = []
```
Comment (codyc1515) [inline custom_components/contact_energy/sensor.py:256]: What's happening here? Identation does not match.

```
Logger: homeassistant.config
Source: loader.py:737
First occurred: 2:12:55 PM (1 occurrences)
Last logged: 2:12:55 PM

Platform error: sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/loader.py", line 728, in get_platform
    cache[full_name] = self._import_platform(platform_name)
  File "/usr/src/homeassistant/homeassistant/loader.py", line 745, in _import_platform
    return importlib.import_module(f"{self.pkg_path}.{platform_name}")
  File "/usr/local/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 879, in exec_module
  File "<frozen importlib._bootstrap_external>", line 1017, in get_code
  File "<frozen importlib._bootstrap_external>", line 947, in source_to_code
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/config/custom_components/contact_energy/sensor.py", line 256
    yesterdaysData = []
IndentationError: unexpected indent

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config.py", line 923, in async_process_component_config
    platform = p_integration.get_platform(domain)
  File "/usr/src/homeassistant/homeassistant/loader.py", line 737, in get_platform
    raise ImportError(
ImportError: Exception importing custom_components.contact_energy.sensor

```

**Review by codyc1515 — COMMENTED** [review]
As it stands, the plug-in can not be loaded due to errors.

File: custom_components/contact_energy/sensor.py [hunk 2]
```diff
@@ -292,7 +270,6 @@ def update(self):
                 self._state_attributes['previous_day'] = yesterdaysData
         else:
             _LOGGER.error('Unable to log in!')
-
     def make_attribute(self, response, value):
```
Comment (codyc1515) [inline custom_components/contact_energy/sensor.py:273]: Why is make_attribute defined twice?

File: custom_components/contact_energy/sensor.py [hunk 3]
```diff
@@ -305,7 +282,6 @@ def make_attribute(self, response, value):
             else:
                 newPoint['price'] = 0
             return newPoint
-
```
Comment (codyc1515) [inline custom_components/contact_energy/sensor.py:308]: There shouldn't be any comment marks (''') at the end of the file as there is no starting comment marks.

File: custom_components/contact_energy/sensor.py (hunk 1 above)
Comment (AprilHickson) [reply custom_components/contact_energy/sensor.py:256]: Sorry, during getting this to work I removed all this commented out code, and just before I contacted you I put it back in. but it looks like I only copied half of it. My bad

File: custom_components/contact_energy/sensor.py (hunk 2 above)
Comment (AprilHickson) [reply custom_components/contact_energy/sensor.py]: See my above comment, I messed up adding back in the commented out code.

File: custom_components/contact_energy/sensor.py (hunk 3 above)
Comment (AprilHickson) [reply custom_components/contact_energy/sensor.py:308]: See my above comment, I messed up adding back in the commented out code.

**Comment by AprilHickson** [comment]
Hey Cody, sorry about that comment issue, I should have double checked before contacting you, sorry about that.

This code is what I have put together, to get the data flowing into my instance, and should work. It has some refactoring work to be done, naturally but I'll do that once you can confirm it's loading historical usage data for you.

**Comment by codyc1515** [comment]
Brilliant - thank you. This works excellently!

**Pull request merged**